File: falleaves-server/auth/src/main/java/cn/falleaves/auth/controller/AuthController.java
package cn.falleaves.auth.controller;

import cn.falleaves.auth.model.dto.UserLoginDTO;
import cn.falleaves.auth.model.vo.UserLoginVO;
import cn.falleaves.auth.model.vo.UserRegisterVO;
import cn.falleaves.auth.service.impl.AuthService;
import cn.falleaves.common.ResponseData;
import cn.falleaves.common.ReturnValue;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;


@RestController
@RequestMapping("/auth")
public class AuthController {

    @Autowired
    private AuthService authService;

    @RequestMapping("/login")
    public ResponseData<? extends String> login(UserLoginVO userLoginVO) {

        if(null == userLoginVO.getIdentifier() || userLoginVO.getIdentifier().equals("")) {
            return ResponseData.New(ReturnValue.IDENTIFIER_EMPTY);
        }

        if(null == userLoginVO.getCredential() || userLoginVO.getCredential().equals("")) {
            return ResponseData.New(ReturnValue.CREDENTIAL_EMPTY);
        }

        try{
            boolean result = authService.login(new UserLoginDTO(userLoginVO.getIdentifier(), userLoginVO.getCredential()));
            if(!result) {
                return ResponseData.New(ReturnValue.INCORRECT_CREDENTIALS);
            }
            return ResponseData.New(ReturnValue.SUCCESS);
        } catch (Exception e) {
            return ResponseData.New(ReturnValue.INTERNET_SERVER_ERROR);
        }
    }

    @RequestMapping("/register")
    public ResponseData register(UserRegisterVO userRegisterVO) {

        ResponseData rd = new ResponseData();

        if(null == userRegisterVO.getNickname() || userRegisterVO.getNickname().equals("")) {
            return ResponseData.New(401.10, "Nickname is empty");
        }

        
        return rd;
    }

    @RequestMapping("/logout")
    public ResponseData logout() {

        ResponseData rd = new ResponseData();



        return rd;
    }
}

File: falleaves-server/auth/src/main/java/cn/falleaves/auth/controller/AuthController.kt
package cn.falleaves.auth.controller

import cn.falleaves.auth.model.dto.UserLoginDTO
import cn.falleaves.auth.model.vo.UserLoginVO
import cn.falleaves.auth.model.vo.UserRegisterVO
import cn.falleaves.auth.service.impl.AuthService
import cn.falleaves.common.ResponseData
import cn.falleaves.common.ReturnValue
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/auth")
class AuthController {

    @Autowired
    private var authService: AuthService? = null

    @RequestMapping("/login")
    fun login(userLoginVO: UserLoginVO): ResponseData<String> {

        if (null == userLoginVO.identifier || userLoginVO.identifier.equals("")) {
            val rd: ResponseData<String> = ResponseData.new(ReturnValue.IDENTIFIER_EMPTY)
            println(rd.code)
            return rd
        }

        if (null == userLoginVO.credential || userLoginVO.credential.equals("")) {
            return ResponseData.new(ReturnValue.CREDENTIAL_EMPTY)
        }

        return try {
            val result = authService!!.login(UserLoginDTO(userLoginVO.identifier, userLoginVO.credential))
            if (!result) {
                ResponseData.new(ReturnValue.INCORRECT_CREDENTIALS)
            } else ResponseData.new(ReturnValue.SUCCESS)
        } catch (e: Exception) {
            ResponseData.new(ReturnValue.INTERNET_SERVER_ERROR)
        }
    }

    @RequestMapping("/register")
    fun register(userRegisterVO: UserRegisterVO): ResponseData<String> {
        return ResponseData(0.0, "null")
    }

    @RequestMapping("/logout")
    fun logout(): ResponseData<String> {
        return ResponseData(0.0, "null")
    }

}